### PR TITLE
Documentation and test fixes

### PR DESCRIPTION
**Pull request by GiudGiud**
- [x] Need to rework commit history
- [x] Need to pass tests :)

Any opposition to removing the bibliography section in doco pages that do not cite anything?

After this PR and another to address #20522, and then https://github.com/idaholab/moose/pull/19578 as well, we'll have very few deprecation failures within the modules, so people will be able to use --error-deprecated on the combined test suite to find all new instances of deprecation

### CODE REVIEW

File: framework/doc/content/source/variables/MooseVariableFE.md [hunk 1]
```diff
@@ -1 +1,11 @@
 # MooseVariableFE
+
+!syntax description /Variables/MooseVariable
+
+!! MOOSE Documentation Stub: Remove this line when content is added.
```
Comment (cticenhour) [inline framework/doc/content/source/variables/MooseVariableFE.md:5]:
```suggestion
!alert construction title=Undocumented
```

File: framework/doc/content/source/vectorpostprocessors/CSVReader.md [hunk 2]
```diff
@@ -6,8 +6,13 @@ converts each column into a VectorPostprocessor vector. This object uses the
 
 ## Example Input Syntax
 
+In this example, the `example.csv` file containings data for year/month/day is being read by
```
Comment (cticenhour) [inline framework/doc/content/source/vectorpostprocessors/CSVReader.md:9]:
```suggestion
In this example, the `example.csv` file containing data for year/month/day is being read by
```

File: framework/doc/content/source/vectorpostprocessors/LineMaterialRealSampler.md [hunk 3]
```diff
@@ -7,8 +7,21 @@
 This class samples Real material properties for the integration points
 in all elements that are intersected by a user-defined line.
 
+The output to CSV is **by default** ordered as follows:
+
+- rows are ordered by point sampled along the line
+
+- columns are ordered by alphabetical order of the properties sampled. The distance along the sampled line, and the x, y and z coordinates of the sampled points are added to the output as additional columns.
```
Comment (cticenhour) [inline framework/doc/content/source/vectorpostprocessors/LineMaterialRealSampler.md:14]:
```suggestion
- columns are ordered by alphabetical order of the properties sampled. The distance along the sampled line and the x, y and z coordinates of the sampled points are added to the output as additional columns.
```

**Comment by GiudGiud** [comment]
If any of the 8 people requested for review need a chill PR to taper off into the weekend... this is it

**Comment by cticenhour** [comment]
Working on this now....almost done.

**Review by cticenhour — CHANGES_REQUESTED** [review]

File: framework/doc/content/source/outputs/OutputWarehouse.md [hunk 4]
```diff
@@ -1 +1,3 @@
 # OutputWarehouse
+
+!! MOOSE Documentation Stub: Remove this line when content is added.
```
[the remark this answers is not in the document]
Comment (GiudGiud) [reply framework/doc/content/source/outputs/OutputWarehouse.md]: Sure lemme know if you want the rest of the stub. It's very temporary, I'll fill those two last classes soon

**Comment by GiudGiud** [comment]
@cticenhour thanks for the review! I merged all your suggestions in.

File: framework/doc/content/source/outputs/OutputWarehouse.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (cticenhour) [reply framework/doc/content/source/outputs/OutputWarehouse.md]: Leaving the rest of it in should be fine!

**Review by cticenhour — APPROVED** [review]

**Comment by GiudGiud** [comment]
Refs #20540 #18333 #20510 #20522 #15283 #20071 #20535 #17478

**Pull request merged**